homeassistant/components/remootio/utils.py

```diff
@@ -0,0 +1,99 @@
+"""Utility methods for the Remootio integration."""
+from __future__ import annotations
+
+import asyncio
+import logging
+from logging import Logger
+
+from aioremootio import ConnectionOptions, LoggerConfiguration, RemootioClient
+from aioremootio.enums import State
+import async_timeout
+
+from homeassistant import core
+from homeassistant.helpers import aiohttp_client
+
+from .const import EXPECTED_MINIMUM_API_VERSION, REMOOTIO_DELAY, REMOOTIO_TIMEOUT
+from .exceptions import (
+    UnsupportedRemootioApiVersionError,
+    UnsupportedRemootioDeviceError,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def _wait_for_connected(remootio_client: RemootioClient) -> bool:
+    while not remootio_client.connected:
+        await asyncio.sleep(REMOOTIO_DELAY)
+
+    return remootio_client.connected
```

Comment:
This can not return anything but True or an exception, so just drop the return value from the function.

Reply:
I don't understand. Why an exception?

I mean that this method returns the value of `remootio_client.connected` is needed because the invocation of the methods `utils._check_sensor_installed(RemootionClient)` and `utils._check_api_version(RemootionClient)` makes only a sense when the the value of `remootio_client.connected` is `True`.

Reply:
You have an while loop above ensuring that connected is always true when you return.

Reply:
Only way the function could do anything else is if the connected property throws an exception, or of something triggers a timeout or task cancellation.